Add macro `do_readme_rmd()`

pat-s opened the pull request:
fixes #65

krlmlr commented on R/macro-readme-rmd.R:
```diff
@@ -0,0 +1,70 @@
+  #' 1. [step_setup_ssh()] in the `"before_deploy"` to setup
+  #'    the upcoming deployment
+  get_stage("before_deploy") %>%
+    add_step(step_setup_ssh(name = ssh_key_name))
```
```suggestion
    add_step(step_setup_ssh(name = !!enquo(ssh_key_name)))
```

krlmlr commented on R/macro-readme-rmd.R:
```diff
@@ -0,0 +1,70 @@
+#' @inheritParams step_setup_ssh
+#' @inheritParams step_setup_push_deploy
+#' @inheritParams step_do_push_deploy
+#' @param path,branch By default, this macro deploys the `"master"` branch
```
Is gh-pages really an alternative?

pat-s replied:
Not really, thanks.

pat-s commented:
@krlmlr 

- Added a section about the macro to the "Getting started" vignette
- added a lifecycle badge including the framework via `usethis::use_lifecycle()`

pat-s commented:
@krlmlr can we merge this?
